module/gui/LinkDock.py

```diff
@@ -19,38 +19,210 @@
 from PyQt4.QtCore import *
 from PyQt4.QtGui import *
 
+import logging
+from os.path import join
+from module.gui.Tools import whatsThisFormat
+
 class NewLinkDock(QDockWidget):
-    def __init__(self):
-        QDockWidget.__init__(self, "New Links")
+    def __init__(self, appIconSet):
+        self.paintEventSignal = False
+        QDockWidget.__init__(self, _("Add Links"))
+        self.log = logging.getLogger("guilog")
+        self.appIconSet = appIconSet
+        
+        self.geo = None
+        self.paintEventCounter = int(0)
+        
+        self.textChangeCount = 0
+        self.undo = False
+        self.undoText = None
+        
         self.setObjectName("New Links Dock")
+        self.setWindowIcon(QIcon(join(pypath, "icons", "logo.png")))
         self.widget = NewLinkWindow(self)
+        self.setFilterBtnText()
+        whatsThis = (self.widget.filter.text(), _("Extracts the URLs from the pasted/dropped text and removes duplicates."))
+        self.widget.filter.setWhatsThis(whatsThisFormat(*whatsThis))
+        self.defaultSettings()
         self.setWidget(self.widget)
-        self.setAllowedAreas(Qt.RightDockWidgetArea|Qt.LeftDockWidgetArea)
+        self.setAllowedAreas(Qt.RightDockWidgetArea)
         self.hide()
     
     def slotDone(self):
-        text = str(self.widget.box.toPlainText())
+        text = unicode(self.widget.box.toPlainText())
         lines = text.splitlines()
-        self.emit(SIGNAL("done"), lines)
+        if not lines:
+            self.widget.slotMsgShow("<b>" + "&nbsp;" + "&nbsp;" + _("Error, no URLs given.") + "</b>")
+            return
+        queue = self.widget.destQueue.isChecked()
+        self.emit(SIGNAL("done"), lines, queue)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['SIGNAL' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)